### PR TITLE
Update version to 2.1.2

### PR DESCRIPTION
**Pull request by rht**
I included #1757, so it must be merged first.

### CODE REVIEW

**Comment by tpike3** [comment]
Can you fix the codespell error on line 41

**Review by Corvince — DISMISSED** [review]
This won't be publishable to PyPi again because of the GitHub dependency of mesa-viz-tornado. We have to package this up first

**Comment by Corvince** [comment]
But thanks a lot for putting this together @rht

**Comment by rht** [comment]
Good catch. I was too hasty, because it has been days without progress.

File: HISTORY.rst
Comment (rht) [inline HISTORY.rst:41]: Note: I copy pasted this from the PR.

**Comment by rht** [comment]
> This won't be publishable to PyPi again because of the GitHub dependency of mesa-viz-tornado. We have to package this up first

I don't have permission to do so. Please don't delay this any longer.

**Comment by Corvince** [comment]
What kind of permissions do you mean? There isn't any package on PyPi named `mesa-viz-tornado` yet, so theoretically anyone can submit such package. I think you also already have some knowledge about packaging a python package, no? So maybe you would be willing to publish? You can also first test on test.pypi.org if everything works. 

Otherwise I myself won't have much time until next week. Don't know how it is with @tpike3 or @jackiekazil

**Comment by Corvince** [comment]
I mean we have to somewhat wait for @tpike3 or @jackiekazil anyhow, because they have to finally publish the mesa package. So they might be willing to do the same with mesa-viz-tornado.

**Comment by rht** [comment]
> There isn't any package on PyPi named mesa-viz-tornado yet, so theoretically anyone can submit such package.

You already know yourself that if it is not published by Mesa's PyPI account, it is bound to draw suspicion.

**Comment by tpike3** [comment]
Just to recap.. 
- we want to release Mesa 2.1.2 today (9/20)
- we need to package up mesa-viz-tornado #1746 but it needs to be packaged and released under Mesa so not to be considered suspicious, but the the plan is to do that later (not today)

**Comment by Corvince** [comment]
> > There isn't any package on PyPi named mesa-viz-tornado yet, so theoretically anyone can submit such package.
> 
> You already know yourself that if it is not published by Mesa's PyPI account, it is bound to draw suspicion.

There is no Mesa PyPi account. Previous versions of mesa have been published by @jackiekazil account. @tpike3 would have the same right. Mesa-geo is either published by myself or @wang-boyu. I wouldn't know **who** would be suspicious if you published that package @rht . After all you are probably the most active contributor at the moment and member of the core team.

**Comment by jackiekazil** [comment]
I can publish right after work today.

**Comment by jackiekazil** [comment]
Catching up on where we are ... please let me know if I need to correct anything. 

* This is blocked by needing mesa-viz-tornado release. 
  * I created a history ticket - does that need to be done as well -- https://github.com/projectmesa/mesa-viz-tornado/issues/29
   * I can do a release of this for speed's sake - we should talk about governance around releases. We discussed this in the past, but we have yet to conclude. (cc @tpike3 for next agenda)
   * readme - we can skip a history for now, but we need a readme; otherwise, people will wonder what is happening. Is there anything else that needs to happen? 
 
Are we imagining the process for now being release mesa-viz-tornado, then Mesa? 
 
 * Release of mesa -- 
   * We should update this requirement before the release -- https://github.com/projectmesa/mesa/blob/main/setup.py#L11

**Comment by rht** [comment]
I think it's fine to not have the history file for mesa-viz-tornado, since technically it is no longer going to be developed anyway.

**Comment by rht** [comment]
I have updated the changelog to include #1813.

**Comment by tpike3** [comment]
@rht are you going to remove the requires from github--- https://github.com/projectmesa/mesa/blob/main/setup.py#L11 or do you want me to so we can release to PyPI?

**Comment by rht** [comment]
You can do it, to save a roundtrip.

**Comment by Corvince** [comment]
> @rht are you going to remove the requires from github--- https://github.com/projectmesa/mesa/blob/main/setup.py#L11 or do you want me to so we can release to PyPI?

Wait, but we can't just do that - that would would be a breaking change. Everyone's visualization would break.

**Comment by rht** [comment]
@tpike3 is going to replace it with a PyPI release version instead of Git version: https://github.com/projectmesa/mesa/pull/1811#issuecomment-1727376679

**Comment by Corvince** [comment]
Ok I wasn't sure from the wording and there isn't a package yet.

**Comment by jackiekazil** [comment]
Quick read me for mesa-viz-tornado? 
I would do it, but know less about it than you all. It dosn't have to be fancy, I would cover the following -- 

TItle 
1 Paragraph description
Possible things to include - what is it? Why it exists? who would use it? what is the future state?

**Comment by jackiekazil** [comment]
Suggested pattern for release -- 

- [ ] Update readme in mesa-viz-tornado, release to pypi
- [ ] Adjust setup.py in this repo to point to pypi, not the repo 
- [ ] Update history in this repo
- [ ] Release this repo

File: setup.py
Comment (rht) [inline setup.py:11]: Syntax error.

Comment (tpike3) [reply setup.py]: Sorry about that, that's what I get for being lazy and seeing what it was like to edit the file in the browser
 now fixed

**Review by jackiekazil — APPROVED** [review]
LGTM

**Comment by jackiekazil** [comment]
Merging for a release.

**Comment by jackiekazil** [comment]
Released - https://pypi.org/project/Mesa/2.1.2/

**Pull request merged**
